src/libraries/System.Net.Http/ref/System.Net.Http.cs

```diff
@@ -212,13 +214,45 @@ public partial class HttpRequestMessage : System.IDisposable
         public System.Net.Http.HttpContent? Content { get { throw null; } set { } }
         public System.Net.Http.Headers.HttpRequestHeaders Headers { get { throw null; } }
         public System.Net.Http.HttpMethod Method { get { throw null; } set { } }
+        [Obsolete("Use Options instead.")]
         public System.Collections.Generic.IDictionary<string, object?> Properties { get { throw null; } }
+        public HttpRequestOptions Options { get { throw null; } }
         public System.Uri? RequestUri { get { throw null; } set { } }
         public System.Version Version { get { throw null; } set { } }
         public void Dispose() { }
         protected virtual void Dispose(bool disposing) { }
         public override string ToString() { throw null; }
     }
+
+    public readonly struct HttpRequestOptionsKey<TValue>
+    {
+        public HttpRequestOptionsKey(string key) {}
+        public string Key { get { throw null; } }
+    }
+
+    public sealed class HttpRequestOptions : System.Collections.Generic.IDictionary<string, object?>
+    {
+        public HttpRequestOptions() { }
+        public void Add(string key, object? value) { throw null; }
+        public bool ContainsKey(string key) { throw null; }
+        public System.Collections.Generic.ICollection<string> Keys { get { throw null; } }
+        public bool Remove(string key) { throw null; }
+        public bool TryGetValue(string key, out object? value) { throw null; }
+        public System.Collections.Generic.ICollection<object?> Values { get { throw null; } }
+        public object? this[string key] { get { throw null; } set { } }
+        public void Add(System.Collections.Generic.KeyValuePair<string, object?> item) { throw null; }
+        public void Clear() { throw null; }
+        public bool Contains(System.Collections.Generic.KeyValuePair<string, object?> item) { throw null; }
+        public void CopyTo(System.Collections.Generic.KeyValuePair<string, object?>[] array, int arrayIndex) { throw null; }
+        public int Count { get { throw null; } }
+        public bool IsReadOnly { get { throw null; } }
+        public bool Remove(System.Collections.Generic.KeyValuePair<string, object?> item) { throw null; }
+        public System.Collections.Generic.IEnumerator<System.Collections.Generic.KeyValuePair<string, object?>> GetEnumerator() { throw null; }
+        System.Collections.IEnumerator System.Collections.IEnumerable.GetEnumerator() { throw null; }
+        public bool TryGetValue<TValue>(HttpRequestOptionsKey<TValue> key, [MaybeNullWhen(false)] out TValue value) { throw null; }
+        public void Set<TValue>(HttpRequestOptionsKey<TValue> key, TValue value) { throw null; }
```

Comment:
I thought the original plan was to implement the interface explicitly. Did that change?

Reply:
I don't think so. The plan was to expose the two methods on this type and explicitly implement the dictionary contract

Reply:
@marek-safar @stephentoub can we open a discussion to come to an efficient implementation?

Reply:
The implementation that is out there should resolve this right?

Reply:
Nevermind the messages above.  There was a problem with GitHub and the commits to the origin branch were not applying to dotnet/runtime repo.  The implementation should be good now.